src/main/java/com/algorand/algosdk/transaction/MethodCallOption.java

```diff
@@ -0,0 +1,135 @@
+package com.algorand.algosdk.transaction;
+
+import com.algorand.algosdk.abi.Method;
+import com.algorand.algosdk.algod.client.model.TransactionParams;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MethodCallOption {
+    public Long appID;
+    public Method method;
+    public List<AtomicTransactionComposer.MethodArgument> methodArgs;
+    public String sender, rekeyTo;
+    public Transaction.OnCompletion onCompletion;
+    public Byte[] note, lease;
+    public BigInteger fv, lv, fee, flatFee;
+    AtomicTransactionComposer.TransactionSigner signer;
+    TransactionParams suggestedParams;
+
+    public MethodCallOption(Long appID, Method method, List<AtomicTransactionComposer.MethodArgument> methodArgs, String sender,
+                             TransactionParams sp, Transaction.OnCompletion onCompletion, Byte[] note, Byte[] lease,
+                             BigInteger fv, BigInteger lv, BigInteger fee, BigInteger flatFee,
+                             String rekeyTo, AtomicTransactionComposer.TransactionSigner signer) {
+        if (appID == null || method == null || sender == null || onCompletion == null || signer == null || sp == null)
+            throw new IllegalArgumentException("Method call builder error: some required field not added");
+        if (method.args.size() != methodArgs.size())
+            throw new IllegalArgumentException("Method call error: incorrect method arg number provided");
+        this.appID = appID;
+        this.method = method;
+        this.methodArgs = methodArgs;
+        this.sender = sender;
+        this.suggestedParams = sp;
+        this.onCompletion = onCompletion;
+        this.note = note;
+        this.lease = lease;
+        this.fv = fv;
+        this.lv = lv;
+        this.fee = fee;
+        this.flatFee = flatFee;
+        this.rekeyTo = rekeyTo;
+        this.signer = signer;
+    }
+
+    public static class MethodCallOptionBuilder {
```

Comment:
Good choice on making a builder for this, it makes a lot of sense!

I'd suggest that `Builder` would be a better name, so to users it just looks like `MethodCallOption.Builder`